plugins/utils/illuminate.nix

```
@@ -0,0 +1,144 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.illuminate;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
+
+  commonOptions = with helpers.defaultNullOpts; {
+    providers = mkListStr ''["lsp" "treesitter" "regex"]'' ''
+      Provider used to get references in the buffer, ordered by priority.
+    '';
+
+    delay = mkInt 100 ''
+      Delay in milliseconds.
+    '';
+
+    modesDenylist = mkListStr "[]" ''
+      Modes to not illuminate, this overrides `modes_allowlist`.
+      See `:help mode()` for possible values.
+    '';
+
+    modesAllowlist = mkListStr "[]" ''
+      Modes to illuminate, this is overriden by `modes_denylist`.
+      See `:help mode()` for possible values.
+    '';
+
+    providersRegexSyntaxDenylist = mkListStr "[]" ''
+      Syntax to not illuminate, this overrides `providers_regex_syntax_allowlist`.
+      Only applies to the 'regex' provider.
+      Use `:echo synIDattr(synIDtrans(synID(line('.'), col('.'), 1)), 'name')`.
+    '';
+
+    providersRegexSyntaxAllowlist = mkListStr "[]" ''
+      Syntax to illuminate, this is overriden by `providers_regex_syntax_denylist`.
+      Only applies to the 'regex' provider.
+      Use `:echo synIDattr(synIDtrans(synID(line('.'), col('.'), 1)), 'name')`.
+    '';
+
+    underCursor = mkBool true ''
+      Whether or not to illuminate under the cursor.
+    '';
+
+    largeFileCutoff = helpers.mkNullOrOption types.int ''
+      Number of lines at which to use `large_file_config`.
+      The `under_cursor` option is disabled when this cutoff is hit.
+    '';
+
+    minCountToHighlight = mkInt 1 ''
+      Minimum number of matches required to perform highlighting.
+    '';
+  };
+
+  filetypeOptions = {
+    filetypesDenylist = mkListStr ''["dirvish" "fugitive"]'' ''
+      Filetypes to not illuminate, this overrides `filetypes_allowlist`.
+    '';
+
+    filetypesAllowlist = mkListStr "[]" ''
+      Filetypes to illuminate, this is overriden by `filetypes_denylist`.
+    '';
+  };
+in {
+  options.plugins.illuminate = with helpers;
+  with defaultNullOpts;
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "illuminate";
+
+      package = mkPackageOption "illuminate" pkgs.vimPlugins.vim-illuminate;
```

```suggestion
      enable = mkEnableOption "vim-illuminate";

      package = mkPackageOption "vim-illuminate" pkgs.vimPlugins.vim-illuminate;
```

Maybe just here, we can leave the real plugin's name.

---

Well even the github page of the plugin uses both `vim-illuminate` and `illuminate.vim` X)

---

Never really understood why someone uses .vim -nvim or .nvim makes at least some sense, but this is weird, especially when it is mainly neovim plugin. I think it still makes sense to change those two, so they match nixpkgs.